src/store/selectors/activities/index.ts

```diff
@@ -277,13 +359,58 @@ function addMintOperationToActivities(
 
   switch (operation.data.type) {
     case MintType.BASIC_INCOME: {
+      // if cache is present, consider combining this basic income mint
+      // operation with existing activity
+      if (combineActivitiesCache.aggregateBasicIncome) {
+        // check if difference in time is too long before merging
+        const existingActivityId =
+          combineActivitiesCache.aggregateBasicIncome.aggregatedActivityId;
+        const existingActivity = activities.get(existingActivityId);
+        if (!existingActivity) {
```

Comment:
Rather than doing this check I guess I can just store the full activity in the cache.